tests/app/dao/test_provider_details_dao.py

```diff
@@ -105,189 +101,110 @@ def test_update_adds_history(restore_provider_details):
-def test_get_current_sms_provider_returns_correct_provider(restore_provider_details):
-    set_primary_sms_provider('mmg')
-
-    provider = get_current_provider('sms')
-
-    assert provider.identifier == 'mmg'
-
-
-@pytest.mark.parametrize('provider_identifier', ['firetext', 'mmg'])
-def test_get_alternative_sms_provider_returns_expected_provider(notify_db, provider_identifier):
-    provider = get_alternative_sms_provider(provider_identifier)
-    assert provider.identifier != provider
-
-
-def test_switch_sms_provider_to_current_provider_does_not_switch(
-    restore_provider_details,
-    current_sms_provider
-):
-    dao_switch_sms_provider_to_provider_with_identifier(current_sms_provider.identifier)
-    new_provider = get_current_provider('sms')
+    mmg.active = False
+    dao_update_provider_details(mmg)
 
-    assert current_sms_provider.id == new_provider.id
-    assert current_sms_provider.identifier == new_provider.identifier
+    assert not mmg.active
 
 
-def test_switch_sms_provider_to_inactive_provider_does_not_switch(
-    restore_provider_details,
-    current_sms_provider
-):
-    alternative_sms_provider = get_alternative_sms_provider(current_sms_provider.identifier)
-    alternative_sms_provider.active = False
-    dao_update_provider_details(alternative_sms_provider)
-
-    dao_switch_sms_provider_to_provider_with_identifier(alternative_sms_provider.identifier)
-    new_provider = get_current_provider('sms')
+@pytest.mark.parametrize('identifier, expected', [
+    ('firetext', 'mmg'),
+    ('mmg', 'firetext'),
+])
+def test_get_alternative_sms_provider_returns_expected_provider(identifier, expected):
+    assert get_alternative_sms_provider(identifier) == expected
 
-    assert new_provider.id == current_sms_provider.id
-    assert new_provider.identifier == current_sms_provider.identifier
-
-
-def test_toggle_sms_provider_switches_provider(
-    mocker,
-    restore_provider_details,
-    current_sms_provider,
-    sample_user
 
-):
-    mocker.patch('app.provider_details.switch_providers.get_user_by_id', return_value=sample_user)
-    dao_toggle_sms_provider(current_sms_provider.identifier)
-    new_provider = get_current_provider('sms')
+def test_get_alternative_sms_provider_fails_if_unrecognised():
+    with pytest.raises(ValueError):
+        get_alternative_sms_provider('ses')
 
-    old_starting_provider = get_provider_details_by_identifier(current_sms_provider.identifier)
 
-    assert new_provider.identifier != old_starting_provider.identifier
-    assert new_provider.priority < old_starting_provider.priority
+@pytest.mark.parametrize(['starting_priorities', 'expected_priorities'], [
+    ({'mmg': 50, 'firetext': 50}, {'mmg': 40, 'firetext': 60}),
+    ({'mmg': 0, 'firetext': 20}, {'mmg': 0, 'firetext': 30}),  # lower bound respected
+    ({'mmg': 50, 'firetext': 100}, {'mmg': 40, 'firetext': 100}),  # upper bound respected
 
+    # document what happens if they have unexpected values outside of the 0 - 100 range
```

Comment:
Optional: I wonder if instead of `unexpected` we should make clear that unlikely but allowed values as can be set at any value through the admin UI. Otherwise it feels weird to test an unexpected case

Reply:
squashed in to original commit